plugins/woocommerce/src/Admin/API/Reports/TimeInterval.php

```diff
@@ -229,17 +229,14 @@ public static function intervals_between( $start_datetime, $end_datetime, $inter
 
 				return (int) floor( ( (int) $diff_timestamp ) / HOUR_IN_SECONDS ) + 1 + $addendum;
 			case 'day':
-				$end_timestamp      = (int) $end_datetime->format( 'U' );
-				$start_timestamp    = (int) $start_datetime->format( 'U' );
-				$addendum           = 0;
+				$days               = $start_datetime->diff( $end_datetime )->format( '%r%a' );
 				$end_hour_min_sec   = (int) $end_datetime->format( 'H' ) * HOUR_IN_SECONDS + (int) $end_datetime->format( 'i' ) * MINUTE_IN_SECONDS + (int) $end_datetime->format( 's' );
 				$start_hour_min_sec = (int) $start_datetime->format( 'H' ) * HOUR_IN_SECONDS + (int) $start_datetime->format( 'i' ) * MINUTE_IN_SECONDS + (int) $start_datetime->format( 's' );
 				if ( $end_hour_min_sec < $start_hour_min_sec ) {
-					$addendum = 1;
+					$days++;
```

Comment:
I don't quite understand this condition, at-least it was never hit when running the monthly reports.
Is this only relevant when looking at specific times of day? (which I don't think we can do, or at-least not in revenue).
If this logic is specifically for DST, should we add a comment describing it?

Reply:
This is pre-existing logic and unrelated to the DST logic.  I updated where the variable gets added to make it more clear, but logic here remains the same.

You can see similar logic in other time units, but I believe how this works is counting partial days.  For example, the diff of days between 10-01-2022 12:00:00 and 10-02-2022 11:00:00 would be 0 since no full days exist.  Adding the check for end time allows us to count those partial days.

Reply:
Thanks for the extra explanation, that clarifies that.